apps/src/lib/kits/maker/connectToMakerBoard.js

```diff
@@ -1,16 +1,34 @@
 /**
  * @file Serialport scanning and board controller creation for Maker Toolkit.
  */
+import {getStore} from '../../../redux';
 import {findPortWithViableDevice} from './portScanning';
 import CircuitPlaygroundBoard from './CircuitPlaygroundBoard';
+import {
+  startConnecting,
+  reportConnected,
+  reportConnectionError,
+} from './redux';
 
 /**
  * Establish a connection to a maker board and return a controller for it.
  * @returns {Promise.<CircuitPlaygroundBoard>}}
  */
 export default function connectToMakerBoard() {
+  const store = getStore();
+  const dispatch = store.dispatch.bind(store);
+  let board;
   return Promise.resolve()
+      .then(() => dispatch(startConnecting()))
       .then(findPortWithViableDevice)
-      .then(port => new CircuitPlaygroundBoard(port))
-      .then(board => board.connect().then(() => board));
+      .then(foundPort => new CircuitPlaygroundBoard(foundPort))
+      .then(newBoard => board = newBoard)
+      .then(() => board.connect())
+      .then(() => dispatch(reportConnected()))
+      .then(() => board)
```

Comment:
style question: What do you think about consolidating `then` blocks? The advantage is that it self-documents which statements are returning promises versus completing synchronously. The disadvantage is that the way you have it looks much nicer, and is presumably just as readable once you know that `new` and `dispatch` return synchronously. 
```
  return Promise.resolve()
      .then(() => {
        dispatch(startConnecting());
        return findPortWithViableDevice();
      }).then(foundPort => {
        board = new CircuitPlaygroundBoard(foundPort);
        return board.connect();
      }).then(() => {
        dispatch(reportConnected());
        return board;
      });
```